### PR TITLE
Decoder reasigment on error

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: api/transaction.go [hunk 1]
```diff
@@ -148,6 +148,7 @@ func RawToTransaction(logger *zap.Logger, cdc *amino.Codec, in []TxResponse, blo
 		txErr := TxLogError{}
 		err := dec.Decode(&lf)
 		if err != nil {
+			dec = json.NewDecoder(readr) // (lukanus): reassign decoder in case of failure
```
Comment (eesmerdag) [inline api/transaction.go:151]: i am not really sure if we need to see **lukanus** in comment sir

Comment (lukanus) [reply api/transaction.go:151]: That's my commenting style. it is the same as `TODO(person):`  and `BUG(person):`  but in this case without the indicator? How that's better? The example may be when you copy paste this code, somewhere else, or move this to another file. In such case git loses the track of who added this comment so it is much better to reach correct person. 
Also, Rob Pike is doing that as well ;D

Comment (eesmerdag) [reply api/transaction.go:151]: 👍

**Review by eesmerdag — APPROVED** [review]

File: api/transaction.go [hunk 2]
```diff
@@ -148,6 +148,7 @@ func RawToTransaction(logger *zap.Logger, cdc *amino.Codec, in []TxResponse, blo
...
 			if err != nil {
```
Comment (pacmessica) [inline api/transaction.go:152]: unnecessary error block (you're already inside a err != nil)

**Review by pacmessica — COMMENTED** [review]
lgtm

File: api/transaction.go (hunk 2 above)
Comment (pacmessica) [reply api/transaction.go:152]: lol didn't see this was already merged 😓

Comment (lukanus) [reply api/transaction.go:152]: yeah you're right 👍

**Pull request merged**